x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/shared/map_processor_type_to_form.tsx

```diff
@@ -68,116 +75,203 @@ export const mapProcessorTypeToDescriptor: MapProcessorTypeToDescriptor = {
+          )
+        : i18n.translate(
+            'xpack.ingestPipelines.processors.defaultDescription.dateIndexName.indexNamePrefixDefault.noPrefixValueLabel',
+            { defaultMessage: 'with no prefix' }
+          );
+      return i18n.translate('xpack.ingestPipelines.processors.defaultDescription.date_index_name', {
+        defaultMessage:
+          'Adds documents to a time-based index based on the timestamp value in "{field}", {prefix}',
+        values: {
+          field,
+          prefix,
+        },
+      });
+    },
   },
   dissect: {
     FieldsComponent: Dissect,
     docLinkPath: '/dissect-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.dissect', {
       defaultMessage: 'Dissect',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.dissect', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.dissect', {
       defaultMessage: 'Uses dissect patterns to extract matches from a field.',
     }),
+    getDefaultDescription: ({ field }) =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.dissect', {
+        defaultMessage: 'Extracts values from "{field}" that match a dissect pattern',
+        values: {
+          field,
+        },
+      }),
   },
   dot_expander: {
     FieldsComponent: DotExpander,
     docLinkPath: '/dot-expand-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.dotExpander', {
       defaultMessage: 'Dot expander',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.dotExpander', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.dotExpander', {
       defaultMessage:
         'Expands a field containing dot notation into an object field. The object field is then accessible by other processors in the pipeline.',
     }),
+    getDefaultDescription: ({ field }) =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.dot_expander', {
+        defaultMessage: 'Expands "{field}" into an object field',
+        values: {
+          field,
+        },
+      }),
   },
   drop: {
     FieldsComponent: Drop,
     docLinkPath: '/drop-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.drop', {
       defaultMessage: 'Drop',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.drop', {
-      defaultMessage:
-        'Drops documents without returning an error. Used to only index documents that meet specified conditions.',
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.drop', {
+      defaultMessage: 'Drops documents without returning an error.',
     }),
+    getDefaultDescription: ({ if: value }) =>
+      value
+        ? i18n.translate('xpack.ingestPipelines.processors.defaultDescription.drop', {
```

Comment:
```suggestion
        i18n.translate('xpack.ingestPipelines.processors.defaultDescription.drop', {
```

Reply:
The description for Drop wasn't rendering, and I think the errant `?` might be why.

Reply:
See my comment regarding "Fail" below. Would this be a suitable alternative to always display:

"Drops documents without returning an error" 👈🏻 taken from the description provided for the type.